packages/cmf/src/interceptors.js

```diff
@@ -0,0 +1,152 @@
+/* eslint-disable no-console */
+/* eslint-disable no-underscore-dangle */
+const interceptors = [];
+
+/**
+ * @private
+ * this function remove all interceptors. Should be used only in tests.
+ */
+function _clear() {
+	interceptors.length = 0;
+}
+
+/**
+ * @private
+ * isInterceptor do some check on the interceptor: -
+ * - must be an Object
+ * - must have either 'request' or 'response' attribute
+ * - interceptor.request must be a function
+ * - interceptor.response must be a function
+ * - interceptor.requestError must be a function
+ * - interceptor.responseError must be a function
+ * @param {object} interceptor to check
+ * @return {boolean} true if interceptor is compliant with requirements
+ */
+function isInterceptor(interceptor) {
+	if (!interceptor) {
+		return false;
+	}
+	if (typeof interceptor !== 'object') {
+		return false;
+	}
+	if (!interceptor.request && !interceptor.response) {
```

Comment:
I'm not sure this condition is ok.
For example we can have an interceptor that check responseError, and if 401 we redirect. This doesn't imply request or response.